web/src/dropdown_widget.ts

```diff
@@ -371,7 +375,7 @@ export class DropdownWidget {
             this.current_value = value;
         }
 
-        const all_options = this.get_options();
+        const all_options = this.get_options(this.current_value);
```

Comment:
@amanagr can you review this commit?

Reply:
Looks like we also need to modify `type DropdownWidgetOptions` above. LGTM otherwise.

Reply:
#29684 for that change